lib/global_id/global_id.rb

```diff
@@ -16,6 +19,9 @@ def create(model, options = {})
         raise ArgumentError, 'An app is required to create a GlobalID. ' \
           'Pass the :app option or set the default GlobalID.app.'
       end
+    rescue URI::InvalidComponentError
```

Comment:
Yeah, I don't like this blanket rescue. I feel that -- unlike a more obscure "that model name contains an invalid character" sort of error -- for this, it'd be more appropriate to raise an ArgumentError if model_id (or model_name) is blank back here: https://github.com/rails/globalid/blob/master/lib/global_id/uri/gid.rb#L83.

But even there, the error should be of the form "you failed to supply an ID". If we want to instead say "you passed an object that doesn't have an ID", we should be checking for that way back on https://github.com/rails/globalid/blob/master/lib/global_id/uri/gid.rb#L65.

Note that *even then*, it's inappropriate for us to suggest `save`: this is GlobalID, not ActiveRecord::GlobalID.